streaming/src/main/scala/org/apache/spark/streaming/dstream/TransformedDStream.scala

```diff
@@ -39,7 +39,10 @@ class TransformedDStream[U: ClassTag] (
   override def slideDuration: Duration = parents.head.slideDuration
 
   override def compute(validTime: Time): Option[RDD[U]] = {
-    val parentRDDs = parents.map(_.getOrCompute(validTime).orNull).toSeq
+    val parentRDDs = parents.map { parent => parent.getOrCompute(validTime).getOrElse(
```

Comment:
If "can't be computed now" and returns None, here `orNull` should not be used, it will get null and pass by user, which will lead to NPE. For example:

```
inputStream.transform(_.union(rdd))
```

If inputStream returns `None` as what you mean by "can't be computed now", this clause will throw NPE, so we should guard out this situation.